src/gui/dialogs/preferences_dialog.cpp

```diff
@@ -95,6 +97,19 @@ struct hotkey_sort_by_desc
 	}
 	const gui2::tpreferences::t_visible_hotkeys* hotkey_commands_;
 };
+
+template<bool reverse>
+struct hotkey_sort_by_command
+{
+	hotkey_sort_by_command(const gui2::tpreferences::t_visible_hotkeys& l) : hotkey_commands_(&l) {}
+	bool operator()(int lhs, int rhs) const
+	{
+		return reverse ? hotkey::get_names((*hotkey_commands_)[lhs]->command) < hotkey::get_names((*hotkey_commands_)[rhs]->command)
```

Comment:
You can just replace the template<bool reverse> with the template<typename comperator> if you want.